File: spec/cpp_stl_11/test_imports_abs_abs.cpp
#include <boost/test/unit_test.hpp>

#include <imports_abs_abs.h>
#include <imported_and_abs.h>
#include <imported_root.h>

#include <iostream>
#include <fstream>
#include <vector>

BOOST_AUTO_TEST_CASE(test_imports_abs_abs) {
    std::ifstream ifs("src/fixed_struct.bin", std::ifstream::binary);
    kaitai::kstream ks(&ifs);
    imports_abs_abs_t* r = new imports_abs_abs_t(&ks);

    BOOST_CHECK_EQUAL(r->one(), 0x50);
    BOOST_CHECK_EQUAL(r->two()->one(), 0x41);
    BOOST_CHECK_EQUAL(r->two()->two()->one(), 0x43);

    delete r;
}

File: spec/java/src/io/kaitai/struct/spec/TestImportsAbsAbs.java
package io.kaitai.struct.spec;

import io.kaitai.struct.testformats.ImportsAbsAbs;
import io.kaitai.struct.testformats.ImportedAndAbs;
import io.kaitai.struct.testformats.ImportedRoot;
import org.testng.annotations.Test;
import static org.testng.Assert.*;
public class TestImportsAbsAbs extends CommonSpec {

    @Test
    public void testImportsAbsAbs() throws Exception {
        ImportsAbsAbs r = ImportsAbsAbs.fromFile(SRC_DIR + "fixed_struct.bin");

        assertIntEquals(r.one(), 80);
        assertIntEquals(r.two().one(), 65);
        assertIntEquals(r.two().two().one(), 67);
    }
}
